### ISSUE TITLE
Course editing is buggy

### ISSUE THREAD

**Issue by Ascent817** [magnifydev/magnify#28]
1. It's possible to delete the field headings:
![image](https://user-images.githubusercontent.com/86618650/181632118-0b956c64-9df8-4cbf-b004-b99a720059d8.png)
2. When cancel is clicked the contents of the card do not revert until reload.

**Comment by GoogleGenius** [comment]
@Ascent817 The contents of the card don't change back because we completely mess up React's handling of the DOM. Honestly, we should revamp this with the modal/form you suggested earlier.

**Comment by Ascent817** [comment]
I agree. I'm leaning towards a modal, especially since you already implemented one. What do you think?

**Comment by GoogleGenius** [comment]
It's a good idea. We really should refactor things so they're more reusable (dynamic components, better CSS, etc.). I often copy and paste code from other elements to have the same style.

**Comment by GoogleGenius** [comment]
Btw, we need to be able to store login info longer term, I find it annoying that it completely resets and we have to login again after a page reload.

**Comment by Ascent817** [comment]
Store with cookies?

**Comment by GoogleGenius** [comment]
I think so. Then we need cookie warning lol :)

**Comment by Ascent817** [comment]
We would be the only people getting cookies lol

**Comment by GoogleGenius** [comment]
I feel like we should rerender the DOM after clicking Cancel, though that would be a performance hit hmm.
It's better as a modal.

**Comment by GoogleGenius** [comment]
This likely needs to be fixed for teachers/admin
